mz_os.c

```diff
@@ -33,7 +33,9 @@ int32_t mz_path_combine(char *path, const char *join, int32_t max_path) {
         path[max_path - 1] = 0;
     } else {
         mz_path_append_slash(path, max_path, MZ_PATH_SLASH_PLATFORM);
-        strncat(path, join, max_path - path_len);
+        path_len = (int32_t)strlen(path);
+        if (max_path > path_len)
```

Comment:
Wouldn't we want to copy as many characters as possible? Here it only copies if it has enough room for all of it..

Reply:
max_path is the capacity of `path`. path_len is what's already in `path`. So if path_len >= max_path, then path is already full and there is no room for any of it.

I would have omitted this check, except strncat takes a size_t (unsigned), so I worry that if the path is already full then `max_path - path_len - 1` will underflow due to the `- 1`, and strncat will try to copy infinite bytes.

mz_path_append_slash keeps the trailing NUL byte in place. So if we can assume that the caller will never pass in a max_path that is shorter than `path` already is, then we can omit this if statement. Let me know if you want me to make that assumption and remove the if().